### PR TITLE
💥Enable Worker heartbeating, update Core to 44a6576

### LINKED ISSUE #551: [Feature Request] Enable Worker Heartbeating
### Describe the solution you'd like

Worker heartbeating is implemented in Core, need to enable worker heartbeating by allowing users to specify the worker heartbeat interval when initializing Core Runtime

### PR DESCRIPTION
**Pull request by yuandrew**
## What was changed

- Enabled worker heartbeat
- plumb plugin names (both client and worker) to core
- update core to 44a6576
  - 💥temporal_sdk_core_c_bridge.dll got renamed to temporalio_sdk_core_c_bridge.dll

## Why?

New worker heartbeating feature

## Checklist

1. Closes #551

2. How was this tested:
Manually tested

3. Any docs updates needed?

---

> [!NOTE]
> Enables worker heartbeating, plumbs plugin names to Core, updates interop/bindings and build paths, and renames the native bridge library to temporalio_sdk_core_c_bridge across code, CI, and docs.
> 
> - **Runtime/Worker**:
>   - Add `TemporalRuntimeOptions.WorkerHeartbeatInterval` and propagate to Core (`worker_heartbeat_interval_millis`).
>   - Pass plugin names (client + worker) to Core via `TemporalCoreByteArrayRefArray` on client/worker options.
>   - Replace `no_remote_activities` with `TemporalCoreWorkerTaskTypes` in `TemporalCoreWorkerOptions` and populate based on registered work.
>   - `Bridge.Worker` constructor accepts `clientPlugins` and forwards to interop conversion.
> - **Interop/Core bindings**:
>   - Update header path to `sdk-core/crates/sdk-core-c-bridge` and adjust all `DllImport` names and library path to `temporalio_sdk_core_c_bridge`.
>   - Add new interop structs/fields: `TemporalCoreWorkerTaskTypes`, extra fields on client/runtime/worker options.
>   - Rename Rust crate log/targets in telemetry defaults from `temporal_*` to `temporalio_*`.
> - **Build/CI/Packaging**:
>   - Rename emitted artifacts to `temporalio_sdk_core_c_bridge.*`; update `.csproj`, `build/net462/Temporalio.targets`, and GitHub Actions workflow to new paths and Cargo manifest.
>   - Adjust CI matrices/includes and smoke-test config; use Manylinux Alpine/ARM containers.
> - **Docs/Tests**:
>   - Update README and test expectations for new library name and log targets; minor test fixes (unique task queues, metric assertions).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2bbb1f484d14b8aca0d7812def95b15c639b4036. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/Temporalio/Temporalio.csproj [hunk 1]
```diff
@@ -34,17 +34,17 @@
   <!-- Build/copy debug/release bridge DLL -->
 
   <PropertyGroup Condition="$([MSBuild]::IsOSPlatform('Windows'))">
-    <BridgeLibraryFile>temporal_sdk_core_c_bridge.dll</BridgeLibraryFile>
+    <BridgeLibraryFile>temporalio_sdk_core_c_bridge.dll</BridgeLibraryFile>
```
Comment (cretz) [inline src/Temporalio/Temporalio.csproj:37]: This is considered a backwards incompatible change. We have to mark the PR with a 💥 in the title of the PR at least so we know to update release notes.

Comment (cretz) [inline src/Temporalio/Temporalio.csproj:37]: This change was not made to the `PackBridgeRuntimes` from below. Should probably also temporarily enable the `package.yml` GH workflow in this branch to confirm it still works.

File: src/Temporalio/Bridge/OptionsExtensions.cs [hunk 2]
```diff
@@ -549,6 +561,16 @@ public static Interop.TemporalCoreWorkerOptions ToInteropOptions(
             options.WorkflowTaskPollerBehavior ??= new PollerBehavior.SimpleMaximum(options.MaxConcurrentWorkflowTaskPolls);
             options.ActivityTaskPollerBehavior ??= new PollerBehavior.SimpleMaximum(options.MaxConcurrentActivityTaskPolls);
             options.NexusTaskPollerBehavior ??= new PollerBehavior.SimpleMaximum(options.MaxConcurrentNexusTaskPolls);
+            var workerPluginNames = options.Plugins?
+                .Select(p => p.Name)
+                .Where(name => !string.IsNullOrEmpty(name)) ?? Enumerable.Empty<string>();
```
Comment (cretz) [inline src/Temporalio/Bridge/OptionsExtensions.cs]: Pedantic, but I don't think we need the `IsNullOrEmpty` check, it is a required property. If we're concerned about empty plugin names, we should validate that somewhere else, but I don't think we're really that concerned about it.

File: src/Temporalio/Runtime/TemporalRuntimeOptions.cs [hunk 3]
```diff
@@ -18,13 +18,33 @@ public TemporalRuntimeOptions()
...
+        {
+            Telemetry = telemetry;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TemporalRuntimeOptions"/> class.
+        /// </summary>
+        /// <param name="telemetry"><see cref="Telemetry" />.</param>
+        /// <param name="workerHeartbeatInterval">Worker heartbeat interval. If 0 is specified, heartbeat is disabled.</param>
+        public TemporalRuntimeOptions(TelemetryOptions telemetry, TimeSpan workerHeartbeatInterval)
```
Comment (cretz) [inline src/Temporalio/Runtime/TemporalRuntimeOptions.cs]: I don't think you need a constructor for this, workerHeartbeatInterval would be rarely set, they can use the parameterless or post-construction set approaches

File: src/Temporalio/Runtime/TemporalRuntimeOptions.cs [hunk 4]
```diff
@@ -18,13 +18,33 @@ public TemporalRuntimeOptions()
...
 
         /// <summary>
         /// Gets or sets the telemetry options.
         /// </summary>
         public TelemetryOptions Telemetry { get; set; } = new();
 
+        /// <summary>
+        /// Gets or sets the worker heartbeat duration in milliseconds.
+        /// </summary>
+        public TimeSpan WorkerHeartbeatInterval { get; set; } = TimeSpan.FromSeconds(60);
```
Comment (cretz) [inline src/Temporalio/Runtime/TemporalRuntimeOptions.cs]: This should be nullable IMO (i.e. `?`)

File: src/Temporalio/Bridge/OptionsExtensions.cs [hunk 5]
```diff
@@ -33,6 +33,7 @@ public static unsafe Interop.TemporalCoreRuntimeOptions ToInteropOptions(
             return new Interop.TemporalCoreRuntimeOptions()
             {
                 telemetry = scope.Pointer(options.Telemetry.ToInteropOptions(scope)),
+                worker_heartbeat_interval_millis = (ulong)options.WorkerHeartbeatInterval.TotalMilliseconds,
```
Comment (cretz) [inline src/Temporalio/Bridge/OptionsExtensions.cs]: This should support unset (if `0` means unset in bridge, then can convert here from null to unset)

**Comment by cretz** [comment]
It looks like something about the Core upgrade broke the `ExecuteWorkflowAsync_CustomMetrics_FloatsAndDurations` test. I think we have to investigate this (let me know if help is needed here).

File: src/Temporalio/Runtime/TemporalRuntimeOptions.cs
Comment (cretz) [inline src/Temporalio/Runtime/TemporalRuntimeOptions.cs:1]: Note, you'll have to update the `TelemetryFilterOptions` class to account for the crate name changes

File: src/Temporalio/Temporalio.csproj (hunk 1 above)
Comment (yuandrew) [reply src/Temporalio/Temporalio.csproj:37]: Thanks, updated PR title and desc

File: src/Temporalio/Runtime/TemporalRuntimeOptions.cs (hunk 4 above)
Comment (yuandrew) [reply src/Temporalio/Runtime/TemporalRuntimeOptions.cs]: sounds good, added in

File: src/Temporalio/Bridge/OptionsExtensions.cs (hunk 5 above)
Comment (yuandrew) [reply src/Temporalio/Bridge/OptionsExtensions.cs]: converted null and unset to 0, lmk if I misunderstood the comment

File: src/Temporalio/Runtime/TemporalRuntimeOptions.cs
Comment (yuandrew) [reply src/Temporalio/Runtime/TemporalRuntimeOptions.cs:1]: ty, I didn't do a thorough search through the repo to update crate names. Kinda thought AI woulda taken care of that, my bad

File: src/Temporalio/Runtime/TemporalRuntimeOptions.cs (hunk 3 above)
Comment (yuandrew) [reply src/Temporalio/Runtime/TemporalRuntimeOptions.cs]: sounds good, removed

File: src/Temporalio/Bridge/OptionsExtensions.cs (hunk 2 above)
Comment (yuandrew) [reply src/Temporalio/Bridge/OptionsExtensions.cs]: removed

File: .github/workflows/package.yml [hunk 6]
```diff
@@ -4,6 +4,7 @@ on:
     branches:
       - main
       - "releases/*"
+  pull_request:
```
Comment (cretz) [inline .github/workflows/package.yml]: Can you remove this now that you've tested this works correctly?

File: src/Temporalio/Bridge/OptionsExtensions.cs [hunk 7]
```diff
@@ -595,6 +619,11 @@ public static Interop.TemporalCoreWorkerOptions ToInteropOptions(
                     throw new ArgumentException("Unable to get assembly manifest ID for build ID");
                 buildId = entryAssembly.ManifestModule.ModuleVersionId.ToString();
             }
+            var pluginNames = options.Plugins?
+                .Select(p => p.Name)
+                .Where(name => !string.IsNullOrEmpty(name))
```
Comment (cretz) [inline src/Temporalio/Bridge/OptionsExtensions.cs]: As cursor bot mentions above, probably don't need this line

File: src/Temporalio/Runtime/TemporalRuntimeOptions.cs
Comment (cretz) [reply src/Temporalio/Runtime/TemporalRuntimeOptions.cs:1]: All good, I missed this in Ruby which was pointed out to me at https://github.com/temporalio/sdk-ruby/pull/368#discussion_r2590704638 which reminded me to point it out here

File: tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs [hunk 8]
```diff
@@ -3381,6 +3381,7 @@ public async Task ExecuteWorkflowAsync_CustomMetrics_WorksWithPrometheus()
                 // We'll also test the metric prefix
                 Metrics = new() { Prometheus = new(promAddr), MetricPrefix = "foo_" },
             },
+            WorkerHeartbeatInterval = null,
```
Comment (cretz) [inline tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs]: I'm curious why these had to be set to `null` in these tests (just trying to make sure our users won't have to do this in their tests)

**Review by cretz — COMMENTED** [review]
LGTM, only blocking concern is to remove the temporary enablement of the `package.yml` GH workflow

File: src/Temporalio/Temporalio.csproj (hunk 1 above)
Comment (yuandrew) [reply src/Temporalio/Temporalio.csproj:37]: package.yml now passes after a fixup, https://github.com/temporalio/sdk-dotnet/actions/runs/19974335695

File: tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs (hunk 8 above)
Comment (yuandrew) [reply tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs]: ahh this is due to the metrics test failing from the `Worker heartbeating configured for runtime, but server version does not support it` warning message, the solution is either to disable it for now, or we can set the dynamic config flag to support heartbeating

Comment (yuandrew) [reply tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs]: the test fails from seeing an additional log statement

Comment (cretz) [reply tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs]: Hrmm, I don't see where these tests are checking logs. Can you explain a bit more how these tests were failing before setting this to null? I want to make sure our users don't have to do the same thing in the same situations. I am not seeing where the metrics would clash with worker heartbeat metrics, but I may be missing it. If unclear where it is failing, I can help debug.

Comment (cretz) [reply tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs]: Discussed off-PR, provided diff for `TestUtils.cs` to help this situation

**Review by cretz — APPROVED** [review]
